### PR TITLE
Update buildsystem to run build in specific container

### PR DESCRIPTION
**Pull request by maor-lb**

### CODE REVIEW

**Comment by muliby-lb** [comment]
@maor-lb just making sure you are aware, this repo is public and open to the world.
I'm not opposed to the lbyaml v2 changes, but we have to make sure the discovery-client remains buildable without it as well for non-LB consumers.

File: Makefile.container [hunk 1]
```diff
@@ -0,0 +1,14 @@
+include $(WORKSPACE_TOP)/lb_containers/Makefile.container
```
Comment (muliby-lb) [inline Makefile.container:1]: missing copyright and license notice

Comment (muliby-lb) [reply Makefile.container:1]: should error out if WORKSPACE_TOP is not defined

File: Makefile.container_inner [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# Copyright 2016--2022 Lightbits Labs Ltd.
```
Comment (muliby-lb) [inline Makefile.container_inner:1]: isn't this just a copy of the regular Makefile?

File: Makefile [hunk 3]
```diff
@@ -74,7 +74,7 @@ discovery-rpms: build/dist build/discovery-client
 	cp -f ${HOME}/rpmbuild/RPMS/x86_64/discovery-*.rpm build/dist/
 
 discovery-client-debs: discovery-rpms
-	(cd build/dist && sudo alien --to-deb -v -k ${WORKSPACE_TOP}/discovery-client/build/dist/discovery-client*.rpm && sudo chown ${USER}:${USER} ${WORKSPACE_TOP}/discovery-client/build/dist/*.deb)
+	(cd build/dist && sudo alien --to-deb -v -k ${WORKSPACE_TOP}/discovery-client/build/dist/discovery-client*.rpm && sudo chown ${USER}: ${WORKSPACE_TOP}/discovery-client/build/dist/*.deb)
```
Comment (muliby-lb) [inline Makefile:77]: change is good but it should be using $(id -gn) for the group for backward compat

File: Makefile.container [hunk 4]
```diff
@@ -6,9 +6,9 @@ TARGETS := install-discovery-client install-discovery-client-packages build-imag
 all: build/discovery-client
 
 ${TARGETS}: discovery_container
-	${Q}run_in_container -i lbdiscovery:${LBDISCOVERY_CONTAINER_VERSION} -c "${MAKE} -f Makefile.container_inner $@"
```
Comment (muliby-lb) [inline Makefile.container]: unrelated change, should probably be a separate commit

File: Makefile.container_inner (hunk 2 above)
Comment (maor-lb) [reply Makefile.container_inner:1]: > @maor-lb just making sure you are aware, this repo is public and open to the world. I'm not opposed to the lbyaml v2 changes, but we have to make sure the discovery-client remains buildable without it as well for non-LB consumers.

Ack. Do we have CI to test public builds ?

Comment (muliby-lb) [reply Makefile.container_inner:1]: Not yet, it's more of a best effort thing right now. Happy to test it when you tell me the PR is ready for testing.

**Comment by muliby-lb** [comment]
@maor-lb should this PR remain open? wip? draft? closed?

**Comment by maor-lb** [comment]
> @maor-lb should this PR remain open? wip? draft? closed?

I'm closing and opening new PR which remove dockerize dependency for build
